doc/manual/rl-next/filesystem-errors.md

```diff
@@ -0,0 +1,12 @@
+---
+synopsis: wrap filesystem exceptions more correctly
+issues: []
+prs: [11378]
+---
+
+
+With the switch to `std::filesystem` in different places, Nix started to throw `std::filesystem::filesystem_error` in many places instead of its own exceptions.
+
+This lead to no longer generating error traces, for example when listing a non-existing directory, and can also lead to crashes inside the Nix REPL.
```

Comment:
Is this documenting the improvement of this PR, or the regression this PR trying to solve?

Reply:
Clarified.